leaper.py

```diff
@@ -125,35 +127,48 @@ def check_source_submission(self, src_project, src_package, src_rev, target_proj
         package = target_package
 
+        origin = None
+
         if src_srcinfo is None:
             # source package does not exist?
             # handle here to avoid crashing on the next line
             self.logger.warn("Could not get source info for %s/%s@%s" % (src_project, src_package, src_rev))
             return False
 
         if self.ibs and target_project.startswith('SUSE:SLE'):
+            if package in self.lookup_sp3:
+                origin = self.lookup_sp3[package]
+
+            origin_same = True
+            if origin:
+                origin_same = True if origin == 'FORK' else src_project.startswith(origin)
+                self.logger.info("expected origin is '%s' (%s)", origin,
+                                 "unchanged" if origin_same else "changed")
+
             # True or None (open request) are acceptable for SLE.
-            if self._check_factory(target_package, src_srcinfo, 'openSUSE.org:openSUSE:Factory') is not False:
-                self.logger.info('found package in openSUSE.org:openSUSE:Factory')
-                return True
+            self.source_in_factory = self._check_factory(package, src_srcinfo, 'openSUSE.org:openSUSE:Factory')
+            if self.source_in_factory is None:
+                self.pending_factory_submission = True
+            if self.source_in_factory is not False:
+                return self.source_in_factory
 
-            if self._check_factory(target_package, src_srcinfo, 'openSUSE.org:openSUSE:Leap:42.2') is not False:
+            if self._check_factory(package, src_srcinfo, 'openSUSE.org:openSUSE:Leap:42.2') is True:
                 self.logger.info('found package in openSUSE.org:openSUSE:Leap:42.2')
-                return True
 
-            devel_project, devel_package = self.get_devel_project('openSUSE.org:openSUSE:Factory', target_package)
-            if devel_project is not None and devel_package is not None:
-                if self._check_factory(devel_package, src_srcinfo, devel_project) is not False:
+            devel_project, devel_package = self.get_devel_project('openSUSE.org:openSUSE:Factory', package)
+            if devel_project is not None:
+                if devel_package is None:
+                    devel_package = package
+                if self.factory._check_project(devel_project, devel_package, src_srcinfo.verifymd5) == True:
                     self.logger.info('found package in {}/{}'.format(devel_project, devel_package))
                     return True
             else:
-                self.logger.info('no devel project found for {}/{}'.format('openSUSE.org:openSUSE:Factory', target_package))
+                self.logger.info('no devel project found for {}/{}'.format('openSUSE.org:openSUSE:Factory', package))
 
             self.logger.info('sources not found in Factory, Leap:42.2, or devel project')
 
-            return False
+            return origin_same
```

Comment:
During our discussions of workflow I started to wonder what the workflow for ignoring leaper's suggestion would be. I imagine certain requests will intentionally not be on OBS.